### PR TITLE
Update `README.md` Github Actions badge

### PR DESCRIPTION
**Pull request by emabrey**
The former badge URL was generating 500 series errors; so let's swap for a different one.

- [x] I have signed off my commits using `-s` or `Signed-off-by`\* (See: [Contributing § DCO](https://github.com/tenacityteam/tenacity/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))
- [x] I made sure the code compiles on my machine
- [x] I made sure there are no unnecessary changes in the code\*
- [x] I made sure the title of the PR reflects the core meaning of the issue you are solving\*
- [x] I made sure the commit message(s) contain a description and answer the question "Why do those changes fix that particular issue?" or "Why are those changes really necessary as improvements?"\*

### CODE REVIEW

**Review by Semisol — DISMISSED** [review]
I don't see any 500 errors with the original but sure I guess.

**Comment by nbsp** [comment]
Could you squash the commits?

**Comment by emabrey** [comment]
> 
> 
> I don't see any 500 errors with the original but sure I guess.

It was giving an error badge for me. And every few requests it would tell me that the serverless worker had crashed.

**Comment by Semisol** [comment]
> Could you squash the commits?

Yep that is what I would want.

File: README.md [hunk 1]
```diff
@@ -3,7 +3,7 @@
 [![Chat on IRC](https://badgen.net/badge/irc/%23tenacity/green)](https://web.libera.chat/gamja/?channels=#tenacity)
 [![License](https://badgen.net/badge/license/GPLv2/blue)](LICENSE.txt)
 [![Open issues](https://badgen.net/github/open-issues/tenacityteam/tenacity)](https://github.com/tenacityteam/tenacity/issues)
-[![GitHub builds](https://badgen.net/github/checks/tenacityteam/tenacity/master?label=github)](https://github.com/tenacityteam/tenacity/actions?query=branch%3Amaster)
+[![GitHub builds](https://badgen.net/github/status/tenacityteam/tenacity)](https://github.com/tenacityteam/tenacity/actions?query=branch%3Amaster)
```
Comment (nbsp) [inline README.md]: Just noticed that this doesn't actually link to what we want. PRs from branches named `master` will also appear. Here's a fix: 

```suggestion
[![GitHub builds](https://badgen.net/github/status/tenacityteam/tenacity)](https://github.com/tenacityteam/tenacity/actions?query=branch%3Amaster+event%3Apush)
```

**Review by nbsp — CHANGES_REQUESTED** [review]

**Review by nbsp — APPROVED** [review]

**Review by fossdd — APPROVED** [review]

**Pull request merged**